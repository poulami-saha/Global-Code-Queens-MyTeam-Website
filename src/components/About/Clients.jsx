import styles from "./Clients.module.css";
import Typography from "../utility/Typography/Typography";
import verge from "../../assets/logo-the-verge.png";
import jakarta from "../../assets/logo-jakarta-post.png";
import guardian from "../../assets/logo-the-guardian.png";
import techRadar from "../../assets/logo-tech-radar.png";
import gadgetsNow from "../../assets/logo-gadgets-now.png";

const Clients = () => {
  return (
    <section className={styles.container}>
      <Typography elType="h2" className={styles.title}>
        Some of our clients
      </Typography>
      <div className={styles.news}>
        <img src={verge} alt="The Verge" className={styles.logoVerge} />
        <img
          src={jakarta}
          alt="TheJakarta Post"
          className={styles.logoJakarta}
        />
        <img
          src={guardian}
          alt="The Guardian"
          className={styles.logoGuardian}
        />
        <img src={techRadar} alt="techRadar" className={styles.logoTechRadar} />
        <img
          src={gadgetsNow}
          alt="Gadgets Now"
          className={styles.logoGadgets}
        />
      </div>
    </section>
  );
};
export default Clients;
