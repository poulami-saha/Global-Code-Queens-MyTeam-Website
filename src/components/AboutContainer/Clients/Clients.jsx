import styles from "./Clients.module.css";
import { Typography } from "../../utility";

import verge from "../../../assets/logo-the-verge.png";
import jakarta from "../../../assets/logo-jakarta-post.png";
import guardian from "../../../assets/logo-the-guardian.png";
import techRadar from "../../../assets/logo-tech-radar.png";
import gadgetsNow from "../../../assets/logo-gadgets-now.png";

const Clients = () => {
  return (
    <section className={styles.container}>
      <div className={styles.body}>
        <Typography elType="h2" className={styles.title}>
          Some of our clients
        </Typography>
        <div className={styles.news}>
          <img src={verge} alt="The Verge" className={styles.logoVerge} />
          <img
            src={jakarta}
            alt="The Jakarta Post"
            className={styles.logoJakarta}
          />
          <img
            src={guardian}
            alt="The Guardian"
            className={styles.logoGuardian}
          />
          <img src={techRadar} alt="Tech Radar" className={styles.logoTechRadar} />
          <img
            src={gadgetsNow}
            alt="Gadgets Now"
            className={styles.logoGadgets}
          />
        </div>
      </div>
    </section>
  );
};
export default Clients;
