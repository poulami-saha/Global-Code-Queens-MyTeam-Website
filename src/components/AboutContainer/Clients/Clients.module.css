.container {
  background: var(--dark-theme-background);
  height: 632px;
  background-image: url("../../../assets/bg-pattern-about-4.svg");
  background-position: top -100px left -100px;
  background-repeat: no-repeat;  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.body {
  display: block;
  width: 100%;
  container-type: inline-size;
  container-name: body-clients;
}

.title {
  color: #fff;
  margin-bottom: 64px;
  margin-top: 24px;
  text-align: center;
}

.news {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 3.5rem;
}

.logoVerge,
.logoGuardian,
.logoTechRadar {
  height: 1.5rem;
  width: auto;
}

.logoJakarta {
  height: 1.25rem;
  width: auto;
}

.logoGadgets {
  height: 2.5rem;
  width: auto;
}

@media (min-width: 768px) {
  .container {
    height: 308px;
    background-position: top -100px left;
  }

  .body {
    width: calc(100% + 58px + 58px); /* match 39px header/footer gutters */
  }

  .title {
    margin-bottom: 48px;
    margin-top: 0;
  }

  .news {
    flex-direction: row;
    gap: 2.5rem;
    justify-content: space-between;
    max-width: 690px;
    margin-inline: auto;
  }

  .logoVerge,
  .logoGuardian,
  .logoTechRadar {
    height: 1.07rem;
  }

  .logoJakarta {
    height: 0.88rem;
  }

  .logoGadgets {
    height: 1.75rem;
  }
}

@media (min-width: 960px) {
  .container {
    height: 437px;
    background-position: top left;
  }

  .body {
    width: 100%;
  }

  .news {
    flex-wrap: wrap;
    justify-content: space-around;
    max-width: unset;
  }

  .logoVerge,
  .logoGuardian,
  .logoTechRadar {
    height: 1.75rem;
  }

  .logoJakarta {
    height: 1.45rem;
  }

  .logoGadgets {
    height: 2.82rem;
  }
}

/* container queries check the size of an element */
@container body-clients (min-width: 955px) {
  /* remove leftmost and rightmost spacing
    when 5 logos can render on one line */
  .news {
    justify-content: space-between;
  }
}
