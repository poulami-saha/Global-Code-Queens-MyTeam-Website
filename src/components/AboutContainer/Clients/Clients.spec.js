import React from "react";
import { render } from "@testing-library/react";
import Clients from "./Clients";

test("renders NotFound correctly and matches snapshot", async () => {
  const { asFragment, getByText } = render(<Clients />);

  // Snapshot test
  expect(asFragment()).toMatchSnapshot();

  // Content test
  expect(getByText(/Some of our clients/)).toBeInTheDocument();
});
