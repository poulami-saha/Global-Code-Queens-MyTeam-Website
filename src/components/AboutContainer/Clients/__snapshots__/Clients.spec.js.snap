// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Clients correctly and matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h2
      class="title h2"
      id=""
    >
      Some of our clients
    </h2>
    <div
      class="news"
    >
      <img
        alt="The Verge"
        class="logoVerge"
        src="logo-the-verge.png"
      />
      <img
        alt="TheJakarta Post"
        class="logoJakarta"
        src="logo-jakarta-post.png"
      />
      <img
        alt="The Guardian"
        class="logoGuardian"
        src="logo-the-guardian.png"
      />
      <img
        alt="techRadar"
        class="logoTechRadar"
        src="logo-tech-radar.png"
      />
      <img
        alt="Gadgets Now"
        class="logoGadgets"
        src="logo-gadgets-now.png"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`renders Clients correctly and matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h2
      class="title h2"
      id=""
    >
      Some of our clients
    </h2>
    <div
      class="news"
    >
      <img
        alt="The Verge"
        class="logoVerge"
        src="logo-the-verge.png"
      />
      <img
        alt="TheJakarta Post"
        class="logoJakarta"
        src="logo-jakarta-post.png"
      />
      <img
        alt="The Guardian"
        class="logoGuardian"
        src="logo-the-guardian.png"
      />
      <img
        alt="techRadar"
        class="logoTechRadar"
        src="logo-tech-radar.png"
      />
      <img
        alt="Gadgets Now"
        class="logoGadgets"
        src="logo-gadgets-now.png"
      />
    </div>
  </section>
</DocumentFragment>
`;
